File: exercism/javascript/secret-handshake/secret-handshake.js
const { reverse, identity } = require('lodash/fp')

const actions = ['wink', 'double blink', 'close your eyes', 'jump']

class SecretHandshake {
  constructor (n) {
    if (typeof n !== 'number') throw Error('Handshake must be a number')
    this.n = n
  }
  commands () {
    const coms = actions.filter((_, i) => (1 << i & this.n) > 0)
    return ((1 << 4 & this.n) > 0 ? reverse : identity)(coms)
  }
}

module.exports = SecretHandshake

File: exercism/csharp/secret-handshake/SecretHandshake.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class SecretHandshake
{
    static string[] Actions = {
      "wink",
      "double blink",
      "close your eyes",
      "jump"
    };

    public static IEnumerable<string> Commands(int n)
    {
        var actions = Actions.Where((x, i) => (1 << i & n) > 0);
        return (1 << 4 & n) > 0 ? actions.Reverse() : actions;
    }
}
